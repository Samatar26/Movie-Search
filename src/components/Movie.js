import React, { Component } from 'react';
import {
  MovieContainer,
  MovieTitle,
  MoviePoster,
  MovieOverview,
} from './../styles/movie';

class Movie extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <MovieContainer>
        <MovieTitle>{this.props.title}</MovieTitle>
        <MoviePoster
          src={`http://image.tmdb.org/t/p/w342${this.props.poster}`}
        />
        <MovieOverview>
          {this.props.overview.replace(/^(.{100}[^\s]*).*/, '$1')}...
        </MovieOverview>
      </MovieContainer>
    );
  }
}

export default Movie;
