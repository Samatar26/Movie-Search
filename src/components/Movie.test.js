import React from 'react';
import Movie from './../components/Movie';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Movie
        title="Logan"
        overview="Logan is the latest installment in the Logan series of Xmen"
        poster="/9EXnebqbb7dOhONLPV9Tg2oh2KD.jpg"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
