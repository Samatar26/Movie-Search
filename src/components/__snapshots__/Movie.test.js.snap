// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<figure
  className="sc-bwzfXH bnKnTl"
>
  <h3
    className="sc-bxivhb lcNDmV"
  >
    Logan
  </h3>
  <img
    className="sc-htpNat gPdQJU"
    src="http://image.tmdb.org/t/p/w342/9EXnebqbb7dOhONLPV9Tg2oh2KD.jpg"
  />
  <p
    className="sc-ifAKCX jtdfU"
  >
    Logan is the latest installment in the Logan series of Xmen
    ...
  </p>
</figure>
`;
